integr8ly/integreatly-operator
# feat: allow editing of GH secret and sync into SSO

### What:
We can no longer create GitHub OAuth apps for users, but it is needed for GH repo import into CodeReady.
We need to allow customer to specify their own GitHub OAuth app credentials if they wish to use repo import feature.

https://issues.redhat.com/browse/INTLY-4818

### Verification Steps

1) install the operator
2) Ensure secret CR `github-oauth-secret` was created in the operator namespace with dummy values
3) Ensure Role and RoleBinding CRs `github-oauth-secret` were created for dedicated-admin SA in the operator namespace
4) Edit the github credentials in the `github-oauth-secret` secret CR 
5) Verify if the GH credentials were added to IdentityProviders in the KeycloakRealm CR on the rhsso namespace 
eg.: `oc get KeycloakRealm openshift  -n <rhsso-namespace> -o jsonpath='{.spec.realm.identityProviders}'`
6) Ensure the installation CR have a status `SetupGHCredentials` set to true
eg.: `oc get installation example-installation -o jsonpath='{.status.setupGHCredentials}` -n <operator-namespace>
